numba/parfors/parfor_lowering.py

```diff
@@ -174,48 +175,48 @@ def _lower_parfor_parallel(lowerer, parfor):
-                               (types.UniTuple(types.intp, redvar_typ.ndim),
-                                reddtype,
-                                types.DType(reddtype)), {})
-                    typemap[full_func.name] = full_fnty
-                    full_assign = ir.Assign(ir.Global("full", np.full, loc=loc), full_func, loc)
-                    lowerer.lower_inst(full_assign)
+                    full_func_node = pfbdr.bind_global_function(
+                        fobj=np.full,
+                        ftype=get_np_ufunc_typ(np.full),
+                        args=(
+                            types.UniTuple(types.intp, redvar_typ.ndim),
+                            reddtype,
+                            types.DType(reddtype),
+                        ),
+                    )
 
                     # Then create a var with the identify value.
-                    init_val_var = ir.Var(scope, mk_unique_var("init_val"), loc)
-                    init_val_assign = ir.Assign(ir.Const(init_val, loc), init_val_var, loc)
-                    typemap[init_val_var.name] = reddtype
-                    lowerer.lower_inst(init_val_assign)
+                    init_val_var = pfbdr.make_const_variable(
+                        cval=init_val,
+                        typ=reddtype,
+                        name="init_val",
+                    )
 
                     # Then, call np.full with the shape of the reduction array and the identity value.
-                    full_call = ir.Expr.call(full_func, [redshape_var, init_val_var], {}, loc=loc)
-                    lowerer.fndesc.calltypes[full_call] = full_sig
-                    redtoset = ir.Var(scope, mk_unique_var("redtoset"), loc)
-                    redtoset_assign = ir.Assign(full_call, redtoset, loc)
-                    typemap[redtoset.name] = redvar_typ
-                    lowerer.lower_inst(redtoset_assign)
+                    full_call = pfbdr.call(
+                        full_func_node, args=[redshape_var, init_val_var],
+                    )
+
+                    redtoset = pfbdr.assign(
+                        rhs=full_call,
+                        typ=redvar_typ,
+                        name="redtoset",
+                    )
                 else:
-                    redtoset = ir.Var(scope, mk_unique_var("redtoset"), loc)
-                    redtoset_assign = ir.Assign(ir.Const(init_val, loc), redtoset, loc)
-                    typemap[redtoset.name] = reddtype
-                    lowerer.lower_inst(redtoset_assign)
+                    redtoset = pfbdr.make_const_variable(
+                        cval=init_val,
+                        typ=reddtype,
+                        name="redtoset",
+                    )
             else:
                 redtoset = redvar
 
             # For each thread, initialize the per-worker reduction array to the current reduction array value.
             for j in range(get_thread_count()):
```

Comment:
Should that be `thread_count` from the var in the outer scope?

Reply:
I changed it to `thread_count`.